server/src/router/v1/index.ts

```diff
@@ -78,4 +79,113 @@ v1App.openapi(
 	},
 );
 
+// Majors
+const majorSchema = createSelectSchema(majors);
+
+v1App.openapi(
+	createRoute({
+		method: 'get',
+		path: '/majors',
+		tags: ['majors'],
+		summary: 'List all majors',
+		responses: {
```

Comment:
we can actually add another parameter `request` and use zod schema to validate request body and params